### PR TITLE
Improvements to performance for views integration of TripalFields

### PR DESCRIPTION
**Pull request by spficklin**
Issue #620 

## Type(s) of Change(s)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API-specific change (fix or addition to an API function)
- [ ] Updates documentation (inline or markdown files)

## Description

When clearing the cache and reloading the site it can take a very long time to return because views is trying to rebuild it's cache for all of the fields attached to Tripal content types.    

For my small site this PR reduces the recovery time from over 1 minute to about 20 seconds.  

## Testing?

1) Before pulling this branch, go the the Structure > Views page.
2)  Clear the cache ```drush cc all```
3)  Reload the Views page and see how long it takes.
4)  Now pull the branch and repeat steps 2 & 3.  The site should return much faster.

## Additional Notes (if any):

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by spficklin** [comment]
@chunhuaicheng I'm hoping this fix may improve performance on your CGD development site.  Can you give it a try and see if it helps any?

File: tripal/tripal.views.inc [hunk 1]
```diff
@@ -116,60 +118,91 @@ function tripal_views_data_alter(&$data) {
...
+  
+  // Get the bundle details so we only look this up once.
+  $all_bundles = [];
+  $sql = "
+    SELECT TB.name, TV.vocabulary, TT.accession
+    FROM {tripal_bundle} TB
+      INNER JOIN {tripal_term} TT on TT.id = TB.term_id
+      INNER JOIN {tripal_vocab} TV on TV.id = TT.vocab_id
+  ";
+  $results = db_query($sql, [':bundle_name' => $bundle_name]);
```
Comment (almasaeed2010) [inline tripal/tripal.views.inc]: The args array shouldn't exist here I think.

Thanks!

**Comment by almasaeed2010** [comment]
Hello @spficklin 

Here is the avg results from 3 different runs:

|Before|After|
|------|-----|
|35.2s|5.89s|

Awesome!


There seems to be a simple error though:
```
Notice: Undefined variable: bundle_name in tripal_views_data_fields() (line 130 of /Users/Almsaeed/Work/DevSites/Tripal/sites/all/modules/tripal/tripal/tripal.views.inc).
```

I added a code-comment for a fix.

Thanks!

File: tripal/tripal.views.inc (hunk 1 above)
Comment (spficklin) [reply tripal/tripal.views.inc]: Yes ,you are right.

**Comment by spficklin** [comment]
@almasaeed2010 I just pushed a fix that removes that array from the db_query.  Thanks for spotting that one.  Not sure how I missed any warning message about that.  

I also merged 7.x-3.x branch into this so it's up to date.

**Comment by almasaeed2010** [comment]
Great! I'll test again to verify the merge is successful.

**Comment by almasaeed2010** [comment]
Perfect. Approved. Thanks @spficklin

**Review by almasaeed2010 — APPROVED** [review]

**Review by laceysanderson — APPROVED** [review]
Code looks good 👍

**Comment by laceysanderson** [comment]
I had a 2.5X increase in speed with this branch 👍 and confirmed that the default views still work after the cache is cleared. 😀

**Comment by laceysanderson** [comment]
That makes 2 approvals so I'm merging this in 🎉

**Comment by spficklin** [comment]
Thanks @laceysanderson and @almasaeed2010

**Pull request merged**